backend/matchmaker_service.py

```diff
@@ -0,0 +1,65 @@
+import firebase_admin
+from firebase_admin import credentials, firestore
+from game import Game
+from game_engine.common import GameOptions
+from game_engine.firestore import FirestoreDB
+import time
+import threading
+
+_FIRESTORE_PROD_CONF_JSON_PATH = ''
+_TEST_FIRESTORE_INSTANCE_JSON_PATH = '../firebase/stv-game-db-test-4c0ec2310b2e.json'
+json_config_path = _TEST_FIRESTORE_INSTANCE_JSON_PATH
+
+
+
+class MatchmakerService:
+    def __init__(self, json_config_path=json_config_path, region="US", min_players=5, is_test=True):
+        self._gamedb = FirestoreDB(json_config_path=json_config_path)
+        self._min_players = min_players
+        self._region=region
+        self._is_test = is_test
+        self._STOP = False
+        self._daemon_started = False
+
+    def matchmaker_function(self, sleep_seconds=60):
+        print("HELLO")
+        # do some logging here
+        while not self._STOP:
+            #games = self._gamedb.find_matchmaker_games(region=self._region)
+            games = [{"id":"IM A GAME", "count_players":6}, {"id":"IM A GAME TOO", "count_players":4}]
+            if len(games) >= 1:
+                print(games)
+                for game in games:
+                    print(game)
+                    if game["count_players"] >= self._min_players:
+                        options = GameOptions()
+                        g = Game(game_id=game["id"],options=options)
+                        if self._is_test:
+                            #start new process
+                            pass
+                        else:
+                            #start on new GCP instance
+                            pass
+                        #g.play()
+                        # Play the game
+            time.sleep(sleep_seconds)
+        print ("STOPPING")
+
+    def start_matchmaker_daemon(self, sleep_seconds=60):

```

Comment:
In general this looks good but wondering if we actually need a background process for the matchmaker daemon. I believe the plan is to have one matchmaker service running on a dedicated GCP instance per region, so the entire matchmaking algorithm can probably run on a single main thread, with some failsafes for exception handling. Interested in your thoughts.

Reply:
Re: running on a single thread, I suppose we could do that, but then how would we update the stop flag? If we go this route we would only be able to stop the daemon by force quitting from the command line, which could result in an unpredictable state if the daemon is stopped while in the middle of doing something (depending on the final implementation).

Reply:
@dzx11 I think even on the main thread we'd still have a loop that's conditioned on the stop event. It may be helpful to enumerate the specific cases that would result in the stop event being set, then we can determine whether those cases are invoked from a separate thread, process, machine etc.

Reply:
@un7c0rn I think we eventually would want some kind of interface for communicating with all the daemons, including stopping. A master service that would start and coordinate with all the regions' matchmakers. This service would be its own thread and then each region would run a single thread on a separate machine. This is sort of what I was trying to encapsulate here, but it's not very fleshed out right now. I think it'll make more sense when the code is more developed but I'm open to hearing if it's the wrong approach.

Reply:
Note from meeting: "arbiter" for handling stopping, etc is out of scope for MVP, so will remove multithreading from this PR